tests/core/test_block_manager.py

```diff
@@ -241,6 +273,62 @@ def test_swap():
     assert before_gpu_blocks == after_gpu_blocks + len(cpu_blocks)
 
 
+def test_swap_encoder_decoder():
+    block_size = 4
+    num_cpu_blocks = 4
+    num_gpu_blocks = 4
+    block_manager = BlockSpaceManagerV1(block_size,
+                                        num_cpu_blocks,
+                                        num_gpu_blocks,
+                                        watermark=0)
+
+    decoder_prompt, encoder_prompt, seq_group = \
+        create_dummy_prompt_encoder_decoder(
+        "1",
+        decoder_prompt_length=block_size,
+        encoder_prompt_length=block_size)
+    decoder_prompt.status = SequenceStatus.WAITING
+    encoder_prompt.status = SequenceStatus.WAITING
+    block_manager.allocate(seq_group)
+
+    # Emulate a forward pass by appending a single token.
+    # The block manager then knows how many unprocessed
+    # tokens will be written in the next forward pass.
+    token_id = 0
+    decoder_prompt.status = SequenceStatus.RUNNING
+    decoder_prompt.append_token_id(token_id, {token_id: Logprob(0.0)})
+
+    # Swap encoder/decoder seq group from GPU -> CPU.
+    decoder_gpu_blocks = block_manager.get_block_table(decoder_prompt)
+    cross_gpu_blocks = block_manager.get_cross_block_table(seq_group)
+    gpu_blocks = decoder_gpu_blocks + cross_gpu_blocks
+    assert block_manager.can_swap_out(seq_group)
```

Comment:
Should this check `== AllocStatus.OK`?

Reply:
Good question!

Based on examining can_swap_out()

https://github.com/neuralmagic/nm-vllm/blob/2ced012a3e51a77abbbab2268d88730fdffa4a3f/vllm/core/block_manager_v1.py#L580

as well as can_swap_in()

https://github.com/neuralmagic/nm-vllm/blob/2ced012a3e51a77abbbab2268d88730fdffa4a3f/vllm/core/block_manager_v1.py#L507

my verdict is that only can_swap_in() requires comparison against the OK enumerated value, whereas can_swap_out() returns plain old bool

Reply:
Kind of interesting that it is set up that way lol. It seems like the original writer choose to use bool when at all possible, but feel back on the enum only was >2 options were absolutely necessary, maybe?